File: wlr/OGUR/OGUR/Statuses/StatusPool.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OGUR.Creatures;

namespace OGUR.Statuses
{
    public class StatusPool
    {
        private List<IStatus> _statuses = new List<IStatus>();

        public bool CanMove()
        {
            for (int ii = 0; ii < _statuses.Count(); ii++)
            {
                if (_statuses[ii].StopMovement())
                {
                    return false;
                }
            }
            return true;
        }

        public bool CanAttack()
        {
            for (int ii = 0; ii < _statuses.Count(); ii++)
            {
                if (_statuses[ii].StopAttack())
                {
                    return false;
                }
            }
            return true;
        }

        public bool WillHitAnything()
        {
            for (int ii = 0; ii < _statuses.Count(); ii++)
            {
                if (_statuses[ii].HitAnything())
                {
                    return true;
                }
            }
            return false;
        }

        public void Add(IStatus status)
        {
            _statuses.Add(status);
        }

        public void Update()
        {
            for (int ii = 0; ii < _statuses.Count(); ii++)
            {
                _statuses[ii].Update();
                if (!_statuses[ii].IsActive())
                {
                    _statuses.Remove(_statuses[ii]);
                    ii--;
                }
            }
        }

        public void PassOn(ICreature target,StatusComponent componentType)
        {
            for (int ii = 0; ii < _statuses.Count; ii++)
            {
                _statuses[ii].PassOn(target, componentType);
            }
        }
    }
}

File: proj/src/aigilas/statuses/StatusPool.java
package aigilas.statuses;

import aigilas.creatures.CreatureAction;
import aigilas.creatures.ICreature;
import aigilas.entities.Elements;

import java.util.ArrayList;
import java.util.List;

public class StatusPool {
    private List<IStatus> _statuses = new ArrayList<IStatus>();

    public boolean allows(CreatureAction action) {
        for (int ii = 0; ii < _statuses.size(); ii++) {
            if (_statuses.get(ii).prevents(action)) {
                return false;
            }
        }
        return true;
    }

    public void add(IStatus status) {
        _statuses.add(status);
    }

    public void update() {
        for (int ii = 0; ii < _statuses.size(); ii++) {
            _statuses.get(ii).update();
            if (!_statuses.get(ii).isActive()) {
                _statuses.remove(_statuses.get(ii));
                ii--;
            }
        }
    }

    public void act() {
        for (int ii = 0; ii < _statuses.size(); ii++) {
            _statuses.get(ii).act();
        }
    }

    public void passOn(ICreature target, StatusComponent componentType) {
        for (int ii = 0; ii < _statuses.size(); ii++) {
            _statuses.get(ii).passOn(target, componentType);
        }
    }

    public boolean isElementBlocked(Elements element) {
        for (IStatus status : _statuses) {
            if (status.isElementBlocked(element)) {
                return true;
            }
        }
        return false;
    }
}
